orangecontrib/text/widgets/ownyt.py

```diff
@@ -10,14 +10,14 @@
 from orangecontrib.text.corpus import Corpus
 from orangecontrib.text.nyt import NYT, MIN_DATE
 from orangecontrib.text.widgets.utils import CheckListLayout, DatePickerInterval, QueryBox, \
-    gui_require, OWConcurrentWidget, asynchronous
+    gui_require, asynchronous
 
 
 class IO:
     CORPUS = "Corpus"
 
 
-class OWNYT(OWConcurrentWidget):
+class OWNYT(OWWidget):
```

Comment:
I investigated this a bit and found that the problem is in the following lines:
```py
# This check must take place in the background thread.
if not getattr(instance, self.method_name).running:
    raise StopExecution
```

If the following code executes than the partial result cannot be returned due to this try except:
```py
try:
    result = self.im_func.method(*args, **kwargs)
except StopExecution:
    result = None
```

As I see it we currently have two ways of stopping the execution:

- First, the running method itself can check the result of `should_break()` method and terminate gracefully (without ever raising `StopExecution`).

- Second, when calling the `callback` the wrapper will check if it is still running and terminate by raising `StopExecution` and hence the partial result will be lost.

I assume that what I am experiencing for NY Times are actually race conditions and that's why the partial result is returned only occasionally. If after clicking stop `should_break` will be called before `callback` everything will be fine but if it happens the other way around I will get `None`. For twitter this is less evident since `should_break` is called after every downloaded tweet and `callback` is only called after every 20 downloaded tweets.

In the case of partial results we shouldn't raise StopExecution errors to prevent race conditions. To do so, I see at least two possible strategies:

1. We can let the the called method decide whether and when to stop and remove the StopExecution raising altogether. I am not sure how that will work for topic modelling.

2. The callback decorator could accept optional argument which would tell if partial results are possible and the raise StopExecution only when they are not possible.

What do you suggest?